BCDA-2770: Update README to include Docker install instructions

lianilychee opened the pull request:
<!--

--- PR Hygiene Checklist ---

1. Make sure your branch is named with this format: `user-initials/description-ABC-123`. For example, `jj/add-awesomeness-bcda-99999`
2. Update the PR title: `bcda-99999 Feature: Add Awesomeness`
3. Edit the text below - do not leave placeholders in the text.
4. Add any other details that will be helpful for the reviewers: details description, screenshots, etc
5. Request a review from someone/multiple someones
-->

<!-- Replace xxx with the JIRA ticket number: -->

### Fixes [BCDA-2770](https://jira.cms.gov/browse/BCDA-2770)

<!-- Describe the problem being solved here: -->

### Proposed Changes

- Added Docker to the "Requirements" list
- Added the definition of a Gem file in "Installation"
- Added the importance of using Docker to build, instead of building locally, in "Using Docker for distributed builds"

<!-- List of changes with bullet points here: -->

### Security Implications

<!-- Does the change deal with PII/PHI at all? What should reviewers look for in
terms of security concerns? -->

- [ ] new software dependencies

<!-- If yes, list the new dependencies and briefly note any relevant security impacts -->

- [ ] security controls or supporting software altered

<!-- If yes, what security controls or supporting software are affected? -->

- [ ] new data stored or transmitted

<!-- If yes, what new data are we storing or transmitting? Is the data considered PII/PHI? -->

- [ ] security checklist is completed for this change

<!-- If yes, provide a link to the security checklist in Confluence here. -->

- [ ] requires more information or team discussion to evaluate security implications

<!-- Use this to indicate you're unsure how this change may impact system security
and would like to solicit the team's feedback. Optionally, provide background
information regarding your questions and concerns. -->

Only changes are verbiage-related.

### Acceptance Validation

<!-- Were you able to fully test the acceptance criteria on the related ticket? if not, why not? -->

<!-- Insert screenshots if applicable (drag images here) -->

Additions called out with orange underlines:

<img width="1002" alt="Screen Shot 2020-02-26 at 2 45 39 PM" src="https://user-images.githubusercontent.com/6895034/75386252-f6ffdc00-58a6-11ea-97db-4a4ecc5893b5.png">

### Feedback Requested

<!-- What type of feedback you want from your reviewers? -->

Do you prefer different wording?

msnook commented on README.md:
```diff
@@ -6,25 +6,27 @@ It is assumed that the environment already has these installed:
 * [rbenv](https://github.com/rbenv/rbenv) or [rvm](https://rvm.io/) to install versioned ruby
 * [ruby](https://www.ruby-lang.org/en/) currently using 2.4
 * [jekyll](https://jekyllrb.com/) currently using 3.5.2
+* [Docker](https://docs.docker.com/docker-for-mac/install/) to standardize builds across all contributors' local machines
```
We probably shouldn't assume the developer will be using a Mac.  Maybe include the generic link, similar to what we have in the `bcda-app` repo [here](https://github.com/CMSgov/bcda-app/blob/master/README.md#dependencies).

msnook replied:
We should also include `docker-compose` in this list.  Again, you can refer to the README in the `bcda-app` repo [here](https://github.com/CMSgov/bcda-app/blob/master/README.md#dependencies).

lianilychee commented:
@msnook , edited README.md :

- Modified Docker install link to be generic, instead of mac-specific (link per bcda-app README)
- Added Docker Compose install instruction (link per bcda-app README)
